apps/api/src/lib/coscrad-query-language/aql/compile-aql-filter-block.ts

```diff
@@ -16,13 +16,16 @@ import {
     CoscradSimpleCondition,
 } from '../models/coscrad-filter-condition';
 
+// This is a hack. How should we handle this?
+let varCount = 0;
+
 interface CoscradAqlFilterBlock {
     letStatement?: string;
     filterStatement: string;
     bindVars: Record<string, unknown>;
 }
 
-const buildFieldref = (
+const buildFieldRef = (
```

Comment:
We need to think about how we want to deal with array element references.  Should `[*]` be treated as one level of field nesting? We are already hitting the need for a general way to apply filters to arrays or deal with nested props on object-valued elements of an array-valued nested prop.